src/main/java/com/prueba/springboot2/dao/ClientDAO.java

```diff
@@ -0,0 +1,49 @@
+package com.prueba.springboot2.dao;
```

Comment:
Al extender tu interfaz de ClientRepository de JpaRepository<T, ID> ya no te hace falta crear esta clase a excepcion del update, el cual no viene out of the box.